### ISSUE TITLE
Resources without a data_layer will fail to compile with AshUUID

### ISSUE THREAD

**Issue by sezaru** [zoonect-oss/ash_uuid#3]
I have some resources that encapsulates some Stripe APIs like this one:

```elixir
defmodule FeedbackCupcake.Payments.SetupIntent do
  @moduledoc false

  use Ash.Resource,
    extensions: [AshUUID]

  code_interface do
    define_for FeedbackCupcake.Payments

    define :new
  end

  attributes do
    uuid_attribute :id

    ...
  end

  actions do
    alias FeedbackCupcake.Payments.SetupIntent.Actions

    defaults [:create]

    create :new do
      ...
    end
  end
end
```

This will fail to compile with this error:

```elixir
== Compilation error in file lib/feedback_cupcake/payments/setup_intent.ex ==
** (RuntimeError) Exception in transformer AshUUID.Transformers.PostgresMigrationDefaults on FeedbackCupcake.Payments.SetupIntent: 

no function clause matching in Keyword.merge/2
    (elixir 1.15.4) lib/keyword.ex:979: Keyword.merge([id: "fragment(\"uuid_generate_v7()\")"], nil)
    (ash_uuid 0.3.0) lib/ash_uuid/transformers/postgres_migration_defaults.ex:56: AshUUID.Transformers.PostgresMigrationDefaults.transform/1
    (spark 1.1.22) lib/spark/dsl/extension.ex:679: anonymous fn/4 in Spark.Dsl.Extension.run_transformers/4
    (elixir 1.15.4) lib/enum.ex:4830: Enumerable.List.reduce/3
    (elixir 1.15.4) lib/enum.ex:2564: Enum.reduce_while/3
    /run/host/home/sezdocs/projects/feedback_cupcake/lib/feedback_cupcake/payments/setup_intent.ex:1: (file)
    (stdlib 5.0.2) erl_eval.erl:750: :erl_eval.do_apply/7
```

Seems like if fails because it tries to create a migration for it even thought it doesn't use postgres.

**Comment by sezaru** [comment]
Just for completeness, this also happens when using it with `embedded` data layers like this one:

```elixir
defmodule FeedbackCupcake.Feedbacks.Student.Image do
  @moduledoc false

  use Ash.Resource,
    data_layer: :embedded,
    extensions: [AshUUID]

  attributes do
    attribute :uuid, :uuid, allow_nil?: false

    ...
  end

  actions do
    ...
  end
end
```

This will give the following error:

```elixir
== Compilation error in file lib/feedback_cupcake/feedbacks/student/image.ex ==
** (RuntimeError) Exception in transformer AshUUID.Transformers.PostgresMigrationDefaults on FeedbackCupcake.Feedbacks.Student.Image: 

no function clause matching in Keyword.merge/2
    (elixir 1.15.4) lib/keyword.ex:979: Keyword.merge([], nil)
    (ash_uuid 0.3.0) lib/ash_uuid/transformers/postgres_migration_defaults.ex:56: AshUUID.Transformers.PostgresMigrationDefaults.transform/1
    (spark 1.1.22) lib/spark/dsl/extension.ex:679: anonymous fn/4 in Spark.Dsl.Extension.run_transformers/4
    (elixir 1.15.4) lib/enum.ex:4830: Enumerable.List.reduce/3
    (elixir 1.15.4) lib/enum.ex:2564: Enum.reduce_while/3
    /run/host/home/sezdocs/projects/feedback_cupcake/lib/feedback_cupcake/feedbacks/student/image.ex:1: (file)
    (stdlib 5.0.2) erl_eval.erl:750: :erl_eval.do_apply/7
```

**Comment by moissela** [comment]
This issue should be resolved in v0.4.0 just released.

Thanks for the feedback @sezaru and let me know if it works as expected 😉
